cli/command/service/update_test.go

```diff
@@ -120,6 +120,43 @@ func TestUpdateGroups(t *testing.T) {
 	assert.Equal(t, groups[2], "wheel")
 }
 
+func TestUpdateDNSConfig(t *testing.T) {
+	flags := newUpdateCommand(nil).Flags()
+
+	flags.Set("dns-add", "1.1.1.1")
+	flags.Set("dns-add", "2.2.2.2")
+	flags.Set("dns-rm", "3.3.3.3")
+	flags.Set("dns-add", "4.4.4.4")
+	flags.Set("dns-rm", "2.2.2.2")
```

Comment:
Thanks @dongluochen. Just updated the PR with additional test cases, covering IPv6, incorrect IP address/domain name, and with duplicates. Please take a look and let me know if there are any issues.